tsinfer/inference.py

```diff
@@ -1019,10 +1020,19 @@ def get_ancestors_tree_sequence(self):
         )
 
         flags, times = tsb.dump_nodes()
-        num_pc_ancestors = count_pc_ancestors(flags)
-        # TODO Write out the metadata here etc also
+        pc_ancestors = is_pc_ancestor(flags)
         tables.nodes.set_columns(flags=flags, time=times)
 
+        # Add metadata for any non-PC node, pointing to the original ancestor
+        metadata = []
+        ancestor = 0
+        for is_pc in pc_ancestors:
+            if is_pc:
+                metadata.append(b"")
+            else:
+                metadata.append(self.encode_metadata({"ancestor": ancestor}))
+                ancestor += 1
```

Comment:
Is there a tidier way to do this?

Reply:
Not sure there's much to be done here, this is as tidy as it'll get I'd say.